### PR TITLE
PAS-446 | Fix memory leak in Serilog

### PR DESCRIPTION
**Pull request by jonashendrickx**
### Ticket

- Closes [PAS-446](https://bitwarden.atlassian.net/browse/PAS-446)

### Description

Fix memory leak caused by Serilog.

### Checklist
I did the following to ensure that my changes were tested thoroughly:
- Load tested locally for several hours.

I did the following to ensure that my changes do not introduce security vulnerabilities:
- __

[PAS-446]: https://bitwarden.atlassian.net/browse/PAS-446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Api/Program.cs [hunk 1]
```diff
@@ -95,11 +90,7 @@
                     configuration: ddConfig);
             }
         }
-    },
-    false,
-    // Pass log events down to other logging providers (e.g. Microsoft) after Serilog, so
-    // that they can be processed in a uniform way in tests.
-    true
```
Comment (abergs) [inline src/Api/Program.cs:102]: @jonashendrickx issue: This was added for a reason, if we just remove it we'll break logging in tests as indicated through the comment/git history.

My suggestion was that we set this argument depending on if we're running in env.Prod instead - was there an issue with my suggestion?

Comment (jonashendrickx) [reply src/Api/Program.cs:102]: I just couldn't read the suggestion in the chat - the thread refused to load.

**Review by abergs — COMMENTED** [review]
Looks good - how can we test that the logs are getting passed through during CI/CD? (cc @Tyrrrz)

**Comment by abergs** [comment]
LGTM, but were we able to test it?

**Comment by jonashendrickx** [comment]
> LGTM, but were we able to test it?

The tests passed. I would be surprised the environment variable for that specific step wouldn't work. We might just have to create a separate flag for it later if running tests in Development mode conflicts with something else.

**Comment by Tyrrrz** [comment]
> Looks good - how can we test that the logs are getting passed through during CI/CD? (cc @Tyrrrz)

It seems that they're there

image

**Pull request merged**